eclair-core/src/main/scala/fr/acinq/eclair/wire/ChannelCodecs.scala

```diff
@@ -26,22 +26,26 @@ import fr.acinq.eclair.crypto.ShaChain
 import fr.acinq.eclair.payment.relay.Origin
 import fr.acinq.eclair.transactions.Transactions._
 import fr.acinq.eclair.transactions._
+import fr.acinq.eclair.wire.ChannelCodecs.{localParamsCodec, waitingForRevocationCodec}
 import fr.acinq.eclair.wire.CommonCodecs._
 import fr.acinq.eclair.wire.LightningMessageCodecs._
 import grizzled.slf4j.Logging
 import scodec.bits.BitVector
 import scodec.codecs._
 import scodec.{Attempt, Codec}
-import shapeless.{HNil, ::}
 
-import scala.compat.Platform
 import scala.concurrent.duration._
 
 /**
  * Created by PM on 02/06/2017.
  */
 object ChannelCodecs extends Logging {
 
+  /**
+   * All LN protocol message must be stored as length-delimited, because they may have arbitrary trailing data
+   */
+  def lengthDelimited[T](codec: Codec[T]): Codec[T] = variableSizeBytes(uint16, codec)
```

Comment:
`uint16` is enough for lightning messages, but you're right that `varint` is both more optimal and more future proof. This will remove some factorization between old/new codecs, but I guess that's not really important.

Reply:
Fixed with 0b956a4e2.